// angular-datatable/__ivy_ngcc__/src/components/header.component.js
import { Component, Inject, forwardRef } from '@angular/core';
import { DataTable } from './table.component';
import { HEADER_TEMPLATE } from './header.template';
import { HEADER_STYLE } from "./header.style";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/forms';

function DataTableHeader_button_7_Template(rf, ctx) { if (rf & 1) {
    var _r3 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "button", 10);
    ɵngcc0.ɵɵlistener("click", function DataTableHeader_button_7_Template_button_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r3); var ctx_r2 = ɵngcc0.ɵɵnextContext(); return ctx_r2.dataTable.downloadItems(); });
    ɵngcc0.ɵɵelement(1, "i", 11);
    ɵngcc0.ɵɵelementEnd();
} }
function DataTableHeader_div_9_div_1_Template(rf, ctx) { if (rf & 1) {
    var _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵelementStart(1, "label");
    ɵngcc0.ɵɵelementStart(2, "input", 16);
    ɵngcc0.ɵɵlistener("ngModelChange", function DataTableHeader_div_9_div_1_Template_input_ngModelChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r9); var ctx_r8 = ɵngcc0.ɵɵnextContext(2); return ctx_r8.dataTable.expandColumnVisible = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r4 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r4.dataTable.expandColumnVisible);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r4.dataTable.translations.expandColumn);
} }
function DataTableHeader_div_9_div_2_Template(rf, ctx) { if (rf & 1) {
    var _r11 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵelementStart(1, "label");
    ɵngcc0.ɵɵelementStart(2, "input", 16);
    ɵngcc0.ɵɵlistener("ngModelChange", function DataTableHeader_div_9_div_2_Template_input_ngModelChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r11); var ctx_r10 = ɵngcc0.ɵɵnextContext(2); return ctx_r10.dataTable.indexColumnVisible = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r5 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r5.dataTable.indexColumnVisible);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r5.dataTable.translations.indexColumn);
} }
function DataTableHeader_div_9_div_3_Template(rf, ctx) { if (rf & 1) {
    var _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 15);
    ɵngcc0.ɵɵelementStart(1, "label");
    ɵngcc0.ɵɵelementStart(2, "input", 16);
    ɵngcc0.ɵɵlistener("ngModelChange", function DataTableHeader_div_9_div_3_Template_input_ngModelChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r13); var ctx_r12 = ɵngcc0.ɵɵnextContext(2); return ctx_r12.dataTable.selectColumnVisible = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "span");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r6 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngModel", ctx_r6.dataTable.selectColumnVisible);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(ctx_r6.dataTable.translations.selectColumn);
} }
function DataTableHeader_div_9_div_4_Template(rf, ctx) { if (rf & 1) {
    var _r16 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 17);
    ɵngcc0.ɵɵelementStart(1, "label");
    ɵngcc0.ɵɵelementStart(2, "input", 16);
    ɵngcc0.ɵɵlistener("ngModelChange", function DataTableHeader_div_9_div_4_Template_input_ngModelChange_2_listener($event) { ɵngcc0.ɵɵrestoreView(_r16); var column_r14 = ctx.$implicit; return column_r14.visible = $event; });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "span", 18);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r14 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngModel", column_r14.visible);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("textContent", column_r14.header);
} }
function DataTableHeader_div_9_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 12);
    ɵngcc0.ɵɵtemplate(1, DataTableHeader_div_9_div_1_Template, 5, 2, "div", 13);
    ɵngcc0.ɵɵtemplate(2, DataTableHeader_div_9_div_2_Template, 5, 2, "div", 13);
    ɵngcc0.ɵɵtemplate(3, DataTableHeader_div_9_div_3_Template, 5, 2, "div", 13);
    ɵngcc0.ɵɵtemplate(4, DataTableHeader_div_9_div_4_Template, 4, 2, "div", 14);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.dataTable.expandableRows);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.dataTable.indexColumn);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r1.dataTable.selectColumn);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r1.dataTable.columns);
} }
export var DataTableHeader = (function () {
    function DataTableHeader(dataTable) {
        this.dataTable = dataTable;
        this.columnSelectorOpen = false;
    }
    DataTableHeader.prototype._closeSelector = function () {
        this.columnSelectorOpen = false;
    };
    /** @nocollapse */
    DataTableHeader.ctorParameters = function () { return [
        { type: DataTable, decorators: [{ type: Inject, args: [forwardRef(function () { return DataTable; }),] },] },
    ]; };
DataTableHeader.ɵfac = function DataTableHeader_Factory(t) { return new (t || DataTableHeader)(ɵngcc0.ɵɵdirectiveInject(forwardRef(function () { return DataTable; }))); };
DataTableHeader.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataTableHeader, selectors: [["data-table-header"]], hostBindings: function DataTableHeader_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function DataTableHeader_click_HostBindingHandler() { return ctx._closeSelector(); }, false, ɵngcc0.ɵɵresolveDocument);
    } }, decls: 10, vars: 5, consts: [[1, "data-table-header"], [1, "title", 3, "textContent"], [1, "button-panel"], ["type", "button", 1, "btn", "btn-default", "btn-sm", "refresh-button", 3, "click"], [1, "fa", "fa-refresh", "fa-lg"], ["type", "button", 1, "btn", "btn-default", "btn-sm", "column-selector-button", 3, "click"], [1, "fa", "fa-list", "fa-lg"], ["type", "button", "class", "btn btn-default btn-sm download-button", 3, "click", 4, "ngIf"], [1, "column-selector-wrapper", 3, "click"], ["class", "column-selector-box panel panel-default", 4, "ngIf"], ["type", "button", 1, "btn", "btn-default", "btn-sm", "download-button", 3, "click"], [1, "fa", "fa-download", "fa-lg"], [1, "column-selector-box", "panel", "panel-default"], ["class", "column-selector-fixed-column checkbox", 4, "ngIf"], ["class", "column-selector-column checkbox", 4, "ngFor", "ngForOf"], [1, "column-selector-fixed-column", "checkbox"], ["type", "checkbox", 3, "ngModel", "ngModelChange"], [1, "column-selector-column", "checkbox"], [3, "textContent"]], template: function DataTableHeader_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "h4", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "button", 3);
        ɵngcc0.ɵɵlistener("click", function DataTableHeader_Template_button_click_3_listener() { return ctx.dataTable.reloadItems(); });
        ɵngcc0.ɵɵelement(4, "i", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "button", 5);
        ɵngcc0.ɵɵlistener("click", function DataTableHeader_Template_button_click_5_listener($event) { ctx.columnSelectorOpen = !ctx.columnSelectorOpen; return $event.stopPropagation(); });
        ɵngcc0.ɵɵelement(6, "i", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(7, DataTableHeader_button_7_Template, 2, 0, "button", 7);
        ɵngcc0.ɵɵelementStart(8, "div", 8);
        ɵngcc0.ɵɵlistener("click", function DataTableHeader_Template_div_click_8_listener($event) { return $event.stopPropagation(); });
        ɵngcc0.ɵɵtemplate(9, DataTableHeader_div_9_Template, 5, 4, "div", 9);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("textContent", ctx.dataTable.headerTitle);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵclassProp("active", ctx.columnSelectorOpen);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.dataTable.showDownloadButton);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.columnSelectorOpen);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgForOf, ɵngcc2.CheckboxControlValueAccessor, ɵngcc2.NgControlStatus, ɵngcc2.NgModel], styles: [".data-table-header[_ngcontent-%COMP%] {\n    min-height: 25px;\n    margin-bottom: 10px;\n}\n.title[_ngcontent-%COMP%] {\n    display: inline-block;\n    margin: 5px 0 0 5px;\n}\n.button-panel[_ngcontent-%COMP%] {\n    float: right;\n}\n.button-panel[_ngcontent-%COMP%]   button[_ngcontent-%COMP%] {\n    outline: none !important;\n}\n\n.column-selector-wrapper[_ngcontent-%COMP%] {\n    position: relative;\n}\n.column-selector-box[_ngcontent-%COMP%] {\n    box-shadow: 0 0 10px lightgray;\n    width: 150px;\n    padding: 10px;\n    position: absolute;\n    right: 0;\n    top: 1px;\n    z-index: 1060;\n}\n.column-selector-box[_ngcontent-%COMP%]   .checkbox[_ngcontent-%COMP%] {\n    margin-bottom: 4px;\n}\n.column-selector-fixed-column[_ngcontent-%COMP%] {\n    font-style: italic;\n}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataTableHeader, [{
        type: Component,
        args: [{
                selector: 'data-table-header',
                template: HEADER_TEMPLATE,
                styles: [HEADER_STYLE],
                host: {
                    '(document:click)': '_closeSelector()'
                }
            }]
    }], function () { return [{ type: DataTable, decorators: [{
                type: Inject,
                args: [forwardRef(function () { return DataTable; })]
            }] }]; }, null); })();
    return DataTableHeader;
}());
