// angular-datatable/__ivy_ngcc__/src/components/pagination.component.js
import { Component, Inject, forwardRef } from '@angular/core';
import { DataTable } from './table.component';
import { PAGINATION_TEMPLATE } from './pagination.template';
import { PAGINATION_STYLE } from "./pagination.style";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/forms';
import * as ɵngcc2 from '../utils/min';

var _c0 = function (a0, a1) { return [a0, a1]; };
export var DataTablePagination = (function () {
    function DataTablePagination(dataTable) {
        this.dataTable = dataTable;
    }
    DataTablePagination.prototype.pageBack = function () {
        this.dataTable.offset -= Math.min(this.dataTable.limit, this.dataTable.offset);
    };
    DataTablePagination.prototype.pageForward = function () {
        this.dataTable.offset += this.dataTable.limit;
    };
    DataTablePagination.prototype.pageFirst = function () {
        this.dataTable.offset = 0;
    };
    DataTablePagination.prototype.pageLast = function () {
        this.dataTable.offset = (this.maxPage - 1) * this.dataTable.limit;
    };
    Object.defineProperty(DataTablePagination.prototype, "maxPage", {
        get: function () {
            return Math.ceil(this.dataTable.itemCount / this.dataTable.limit);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTablePagination.prototype, "limit", {
        get: function () {
            return this.dataTable.limit;
        },
        set: function (value) {
            this.dataTable.limit = Number(value); // TODO better way to handle that value of number <input> is string?
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTablePagination.prototype, "page", {
        get: function () {
            return this.dataTable.page;
        },
        set: function (value) {
            this.dataTable.page = Number(value);
        },
        enumerable: true,
        configurable: true
    });
    /** @nocollapse */
    DataTablePagination.ctorParameters = function () { return [
        { type: DataTable, decorators: [{ type: Inject, args: [forwardRef(function () { return DataTable; }),] },] },
    ]; };
DataTablePagination.ɵfac = function DataTablePagination_Factory(t) { return new (t || DataTablePagination)(ɵngcc0.ɵɵdirectiveInject(forwardRef(function () { return DataTable; }))); };
DataTablePagination.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataTablePagination, selectors: [["data-table-pagination"]], decls: 33, vars: 18, consts: [[1, "pagination-box"], [1, "pagination-range"], [3, "textContent"], [1, "pagination-controllers"], [1, "pagination-limit"], [1, "input-group"], [1, "input-group-addon"], ["type", "number", "min", "1", "step", "1", 1, "form-control", 3, "ngModel", "blur", "keyup.enter", "keyup.esc"], ["limitInput", ""], [1, "pagination-pages"], [1, "btn", "btn-default", "pagination-firstpage", 3, "disabled", "click"], [1, "btn", "btn-default", "pagination-prevpage", 3, "disabled", "click"], [1, "pagination-page"], ["type", "number", "min", "1", "step", "1", 1, "form-control", 3, "max", "ngModel", "blur", "keyup.enter", "keyup.esc"], ["pageInput", ""], [1, "btn", "btn-default", "pagination-nextpage", 3, "disabled", "click"], [1, "btn", "btn-default", "pagination-lastpage", 3, "disabled", "click"]], template: function DataTablePagination_Template(rf, ctx) { if (rf & 1) {
        var _r2 = ɵngcc0.ɵɵgetCurrentView();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵelement(3, "span", 2);
        ɵngcc0.ɵɵtext(4, " - ");
        ɵngcc0.ɵɵelement(5, "span", 2);
        ɵngcc0.ɵɵpipe(6, "min");
        ɵngcc0.ɵɵtext(7, " / ");
        ɵngcc0.ɵɵelement(8, "span", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 3);
        ɵngcc0.ɵɵelementStart(10, "div", 4);
        ɵngcc0.ɵɵelementStart(11, "div", 5);
        ɵngcc0.ɵɵelementStart(12, "span", 6);
        ɵngcc0.ɵɵtext(13);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "input", 7, 8);
        ɵngcc0.ɵɵlistener("blur", function DataTablePagination_Template_input_blur_14_listener() { ɵngcc0.ɵɵrestoreView(_r2); var _r0 = ɵngcc0.ɵɵreference(15); return ctx.limit = _r0.value; })("keyup.enter", function DataTablePagination_Template_input_keyup_enter_14_listener() { ɵngcc0.ɵɵrestoreView(_r2); var _r0 = ɵngcc0.ɵɵreference(15); return ctx.limit = _r0.value; })("keyup.esc", function DataTablePagination_Template_input_keyup_esc_14_listener() { ɵngcc0.ɵɵrestoreView(_r2); var _r0 = ɵngcc0.ɵɵreference(15); return _r0.value = ctx.limit; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(16, "div", 9);
        ɵngcc0.ɵɵelementStart(17, "button", 10);
        ɵngcc0.ɵɵlistener("click", function DataTablePagination_Template_button_click_17_listener() { return ctx.pageFirst(); });
        ɵngcc0.ɵɵtext(18, "\u00AB");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "button", 11);
        ɵngcc0.ɵɵlistener("click", function DataTablePagination_Template_button_click_19_listener() { return ctx.pageBack(); });
        ɵngcc0.ɵɵtext(20, "\u2039");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 12);
        ɵngcc0.ɵɵelementStart(22, "div", 5);
        ɵngcc0.ɵɵelementStart(23, "input", 13, 14);
        ɵngcc0.ɵɵlistener("blur", function DataTablePagination_Template_input_blur_23_listener() { ɵngcc0.ɵɵrestoreView(_r2); var _r1 = ɵngcc0.ɵɵreference(24); return ctx.page = _r1.value; })("keyup.enter", function DataTablePagination_Template_input_keyup_enter_23_listener() { ɵngcc0.ɵɵrestoreView(_r2); var _r1 = ɵngcc0.ɵɵreference(24); return ctx.page = _r1.value; })("keyup.esc", function DataTablePagination_Template_input_keyup_esc_23_listener() { ɵngcc0.ɵɵrestoreView(_r2); var _r1 = ɵngcc0.ɵɵreference(24); return _r1.value = ctx.page; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(25, "div", 6);
        ɵngcc0.ɵɵelementStart(26, "span");
        ɵngcc0.ɵɵtext(27, "/");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(28, "span", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(29, "button", 15);
        ɵngcc0.ɵɵlistener("click", function DataTablePagination_Template_button_click_29_listener() { return ctx.pageForward(); });
        ɵngcc0.ɵɵtext(30, "\u203A");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(31, "button", 16);
        ɵngcc0.ɵɵlistener("click", function DataTablePagination_Template_button_click_31_listener() { return ctx.pageLast(); });
        ɵngcc0.ɵɵtext(32, "\u00BB");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.dataTable.translations.paginationRange, ": ");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("textContent", ctx.dataTable.offset + 1);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("textContent", ɵngcc0.ɵɵpipeBind1(6, 13, ɵngcc0.ɵɵpureFunction2(15, _c0, ctx.dataTable.offset + ctx.dataTable.limit, ctx.dataTable.itemCount)));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("textContent", ctx.dataTable.itemCount);
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵtextInterpolate1("", ctx.dataTable.translations.paginationLimit, ":");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngModel", ctx.limit);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("disabled", ctx.dataTable.offset <= 0);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("disabled", ctx.dataTable.offset <= 0);
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵpropertyInterpolate("max", ctx.maxPage);
        ɵngcc0.ɵɵproperty("ngModel", ctx.page);
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵproperty("textContent", ctx.dataTable.lastPage);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("disabled", ctx.dataTable.offset + ctx.dataTable.limit >= ctx.dataTable.itemCount);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("disabled", ctx.dataTable.offset + ctx.dataTable.limit >= ctx.dataTable.itemCount);
    } }, directives: [ɵngcc1.NumberValueAccessor, ɵngcc1.DefaultValueAccessor, ɵngcc1.NgControlStatus, ɵngcc1.NgModel], pipes: [ɵngcc2.MinPipe], styles: [".pagination-box[_ngcontent-%COMP%] {\n    position: relative;\n    margin-top: -10px;\n}\n.pagination-range[_ngcontent-%COMP%] {\n    margin-top: 7px;\n    margin-left: 3px;\n    display: inline-block;\n}\n.pagination-controllers[_ngcontent-%COMP%] {\n    float: right;\n}\n.pagination-controllers[_ngcontent-%COMP%]   input[_ngcontent-%COMP%] {\n    min-width: 60px;\n    \n    text-align: right;\n}\n\n.pagination-limit[_ngcontent-%COMP%] {\n    margin-right: 25px;\n    display: inline-table;\n    width: 150px;\n}\n.pagination-pages[_ngcontent-%COMP%] {\n    display: inline-block;\n}\n.pagination-page[_ngcontent-%COMP%] {\n    width: 110px;\n    display: inline-table;\n}\n.pagination-box[_ngcontent-%COMP%]   button[_ngcontent-%COMP%] {\n    outline: none !important;\n}\n.pagination-prevpage[_ngcontent-%COMP%], .pagination-nextpage[_ngcontent-%COMP%], .pagination-firstpage[_ngcontent-%COMP%], .pagination-lastpage[_ngcontent-%COMP%] {\n    vertical-align: top;\n}\n.pagination-reload[_ngcontent-%COMP%] {\n    color: gray;\n    font-size: 12px;\n}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataTablePagination, [{
        type: Component,
        args: [{
                selector: 'data-table-pagination',
                template: PAGINATION_TEMPLATE,
                styles: [PAGINATION_STYLE]
            }]
    }], function () { return [{ type: DataTable, decorators: [{
                type: Inject,
                args: [forwardRef(function () { return DataTable; })]
            }] }]; }, null); })();
    return DataTablePagination;
}());
