// angular-datatable/__ivy_ngcc__/src/components/row.component.js
import { Component, Input, Inject, forwardRef, Output, EventEmitter } from '@angular/core';
import { DataTable } from './table.component';
import { ROW_TEMPLATE } from './row.template';
import { ROW_STYLE } from "./row.style";
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../utils/hide';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '@angular/forms';

var _c0 = ["dataTableRow", ""];
function DataTableRow_td_6_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 11);
} if (rf & 2) {
    var column_r2 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("textContent", ctx_r3.item[column_r2.property]);
} }
var _c1 = function (a0, a1, a2) { return { column: a0, row: a1, item: a2 }; };
function DataTableRow_td_6_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 12);
} if (rf & 2) {
    var column_r2 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", column_r2.cellTemplate)("ngOutletContext", ɵngcc0.ɵɵpureFunction3(2, _c1, column_r2, ctx_r4._this, ctx_r4.item));
} }
function DataTableRow_td_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "td", 8);
    ɵngcc0.ɵɵtemplate(1, DataTableRow_td_6_div_1_Template, 1, 1, "div", 9);
    ɵngcc0.ɵɵtemplate(2, DataTableRow_td_6_div_2_Template, 1, 6, "div", 10);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r2 = ctx.$implicit;
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵstyleProp("background-color", column_r2.getCellColor(ctx_r0._this, ctx_r0.index));
    ɵngcc0.ɵɵproperty("hide", !column_r2.visible)("ngClass", column_r2.styleClassObject);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !column_r2.cellTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", column_r2.cellTemplate);
} }
var _c2 = function (a0, a1) { return { row: a0, item: a1 }; };
function DataTableRow_tr_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr", 13);
    ɵngcc0.ɵɵelementStart(1, "td");
    ɵngcc0.ɵɵelement(2, "div", 12);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r1 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("colspan", ctx_r1.dataTable.columnCount);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", ctx_r1.dataTable.expandTemplate)("ngOutletContext", ɵngcc0.ɵɵpureFunction2(3, _c2, ctx_r1._this, ctx_r1.item));
} }
var _c3 = function (a0, a1) { return { "fa-caret-right": a0, "fa-caret-down": a1 }; };
export var DataTableRow = (function () {
    function DataTableRow(dataTable) {
        this.dataTable = dataTable;
        this.selectedChange = new EventEmitter();
        this.expandRowChange = new EventEmitter();
        this._this = this; // FIXME is there no template keyword for this in angular 2?
    }
    Object.defineProperty(DataTableRow.prototype, "selected", {
        get: function () {
            return this._selected;
        },
        set: function (selected) {
            this._selected = selected;
            this.selectedChange.emit(selected);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTableRow.prototype, "displayIndex", {
        // other:
        get: function () {
            if (this.dataTable.pagination) {
                return this.dataTable.displayParams.offset + this.index + 1;
            }
            else {
                return this.index + 1;
            }
        },
        enumerable: true,
        configurable: true
    });
    DataTableRow.prototype.getTooltip = function () {
        if (this.dataTable.rowTooltip) {
            return this.dataTable.rowTooltip(this.item, this, this.index);
        }
        return '';
    };
    DataTableRow.prototype.expandRow = function (event) {
        event.stopPropagation();
        this.expanded = !this.expanded;
        this.expandRowChange.emit();
    };
    DataTableRow.prototype.ngOnDestroy = function () {
        this.selected = false;
    };
    /** @nocollapse */
    DataTableRow.ctorParameters = function () { return [
        { type: DataTable, decorators: [{ type: Inject, args: [forwardRef(function () { return DataTable; }),] },] },
    ]; };
    DataTableRow.propDecorators = {
        'item': [{ type: Input },],
        'index': [{ type: Input },],
        'selectedChange': [{ type: Output },],
        'expandRowChange': [{ type: Output },],
    };
DataTableRow.ɵfac = function DataTableRow_Factory(t) { return new (t || DataTableRow)(ɵngcc0.ɵɵdirectiveInject(forwardRef(function () { return DataTable; }))); };
DataTableRow.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataTableRow, selectors: [["", "dataTableRow", ""]], inputs: { item: "item", index: "index" }, outputs: { selectedChange: "selectedChange", expandRowChange: "expandRowChange" }, attrs: _c0, decls: 8, vars: 22, consts: [[1, "data-table-row", 3, "title", "dblclick", "click"], [1, "row-expand-button", 3, "hide", "click"], [1, "fa", "fa-lg", 3, "ngClass"], [1, "index-column", 3, "hide", "textContent"], [1, "select-column", 3, "hide"], ["type", "checkbox", 3, "ngModel", "ngModelChange"], ["class", "data-column", 3, "hide", "ngClass", "background-color", 4, "ngFor", "ngForOf"], ["class", "row-expansion", 4, "ngIf"], [1, "data-column", 3, "hide", "ngClass"], [3, "textContent", 4, "ngIf"], [3, "ngTemplateOutlet", "ngOutletContext", 4, "ngIf"], [3, "textContent"], [3, "ngTemplateOutlet", "ngOutletContext"], [1, "row-expansion"]], template: function DataTableRow_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "tr", 0);
        ɵngcc0.ɵɵlistener("dblclick", function DataTableRow_Template_tr_dblclick_0_listener($event) { return ctx.dataTable.rowDoubleClicked(ctx._this, $event); })("click", function DataTableRow_Template_tr_click_0_listener($event) { return ctx.dataTable.rowClicked(ctx._this, $event); });
        ɵngcc0.ɵɵelementStart(1, "td", 1);
        ɵngcc0.ɵɵlistener("click", function DataTableRow_Template_td_click_1_listener($event) { return ctx.expandRow($event); });
        ɵngcc0.ɵɵelement(2, "i", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelement(3, "td", 3);
        ɵngcc0.ɵɵelementStart(4, "td", 4);
        ɵngcc0.ɵɵelementStart(5, "input", 5);
        ɵngcc0.ɵɵlistener("ngModelChange", function DataTableRow_Template_input_ngModelChange_5_listener($event) { return ctx.selected = $event; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(6, DataTableRow_td_6_Template, 3, 6, "td", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(7, DataTableRow_tr_7_Template, 3, 6, "tr", 7);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.dataTable.getRowColor(ctx.item, ctx.index, ctx._this));
        ɵngcc0.ɵɵclassProp("row-odd", ctx.index % 2 === 0)("row-even", ctx.index % 2 === 1)("selected", ctx.selected)("clickable", ctx.dataTable.selectOnRowClick);
        ɵngcc0.ɵɵproperty("title", ctx.getTooltip());
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("hide", !ctx.dataTable.expandColumnVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(19, _c3, !ctx.expanded, ctx.expanded));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("hide", !ctx.dataTable.indexColumnVisible)("textContent", ctx.displayIndex);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("hide", !ctx.dataTable.selectColumnVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngModel", ctx.selected);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.dataTable.columns);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.dataTable.expandableRows && ctx.expanded);
    } }, directives: [ɵngcc1.Hide, ɵngcc2.NgClass, ɵngcc3.CheckboxControlValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel, ɵngcc2.NgForOf, ɵngcc2.NgIf, ɵngcc2.NgTemplateOutlet], styles: [".select-column[_ngcontent-%COMP%] {\n    text-align: center;\n}\n\n.row-expand-button[_ngcontent-%COMP%] {\n    cursor: pointer;\n    text-align: center;\n}\n\n.clickable[_ngcontent-%COMP%] {\n    cursor: pointer;\n}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataTableRow, [{
        type: Component,
        args: [{
                selector: '[dataTableRow]',
                template: ROW_TEMPLATE,
                styles: [ROW_STYLE]
            }]
    }], function () { return [{ type: DataTable, decorators: [{
                type: Inject,
                args: [forwardRef(function () { return DataTable; })]
            }] }]; }, { selectedChange: [{
            type: Output
        }], expandRowChange: [{
            type: Output
        }], item: [{
            type: Input
        }], index: [{
            type: Input
        }] }); })();
    return DataTableRow;
}());
