// angular-datatable/__ivy_ngcc__/src/components/table.component.js
import { Component, Input, Output, EventEmitter, ContentChildren, ContentChild, ViewChildren } from '@angular/core';
import { DataTableColumn } from './column.component';
import { DataTableRow } from './row.component';
import { defaultTranslations } from './types';
import { drag } from '../utils/drag';
import { TABLE_TEMPLATE } from './table.template';
import { TABLE_STYLE } from "./table.style";
import * as ɵngcc0 from '@angular/core';

var _c0 = ["dataTableExpand"];
function DataTable_data_table_header_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "data-table-header");
} }
function DataTable_th_11_span_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 6);
} if (rf & 2) {
    var column_r6 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("textContent", column_r6.header);
} }
var _c1 = function (a0) { return { column: a0 }; };
function DataTable_th_11_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "span", 19);
} if (rf & 2) {
    var column_r6 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("ngTemplateOutlet", column_r6.headerTemplate)("ngOutletContext", ɵngcc0.ɵɵpureFunction1(2, _c1, column_r6));
} }
function DataTable_th_11_span_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 20);
    ɵngcc0.ɵɵelement(1, "i", 21);
    ɵngcc0.ɵɵelementStart(2, "span", 22);
    ɵngcc0.ɵɵelement(3, "i", 23);
    ɵngcc0.ɵɵelement(4, "i", 24);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r6 = ɵngcc0.ɵɵnextContext().$implicit;
    var ctx_r10 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("hide", column_r6.property === ctx_r10.sortBy);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("hide", column_r6.property !== ctx_r10.sortBy);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("hide", ctx_r10.sortAsc);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("hide", !ctx_r10.sortAsc);
} }
function DataTable_th_11_span_6_Template(rf, ctx) { if (rf & 1) {
    var _r16 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "span", 25);
    ɵngcc0.ɵɵlistener("mousedown", function DataTable_th_11_span_6_Template_span_mousedown_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r16); var column_r6 = ɵngcc0.ɵɵnextContext().$implicit; var _r7 = ɵngcc0.ɵɵreference(1); var ctx_r15 = ɵngcc0.ɵɵnextContext(); return ctx_r15.resizeColumnStart($event, column_r6, _r7); });
    ɵngcc0.ɵɵelementEnd();
} }
function DataTable_th_11_Template(rf, ctx) { if (rf & 1) {
    var _r19 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "th", 13, 14);
    ɵngcc0.ɵɵlistener("click", function DataTable_th_11_Template_th_click_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r19); var column_r6 = ctx.$implicit; var ctx_r18 = ɵngcc0.ɵɵnextContext(); return ctx_r18.headerClicked(column_r6, $event); });
    ɵngcc0.ɵɵpipe(2, "px");
    ɵngcc0.ɵɵtemplate(3, DataTable_th_11_span_3_Template, 1, 1, "span", 15);
    ɵngcc0.ɵɵtemplate(4, DataTable_th_11_span_4_Template, 1, 4, "span", 16);
    ɵngcc0.ɵɵtemplate(5, DataTable_th_11_span_5_Template, 5, 4, "span", 17);
    ɵngcc0.ɵɵtemplate(6, DataTable_th_11_span_6_Template, 1, 0, "span", 18);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var column_r6 = ctx.$implicit;
    ɵngcc0.ɵɵstyleProp("width", ɵngcc0.ɵɵpipeBind1(2, 12, column_r6.width));
    ɵngcc0.ɵɵclassProp("sortable", column_r6.sortable)("resizable", column_r6.resizable);
    ɵngcc0.ɵɵproperty("hide", !column_r6.visible)("ngClass", column_r6.styleClassObject);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngIf", !column_r6.headerTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", column_r6.headerTemplate);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", column_r6.sortable);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", column_r6.resizable);
} }
function DataTable_tbody_12_Template(rf, ctx) { if (rf & 1) {
    var _r24 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "tbody", 26, 27);
    ɵngcc0.ɵɵlistener("selectedChange", function DataTable_tbody_12_Template_tbody_selectedChange_0_listener() { ɵngcc0.ɵɵrestoreView(_r24); var _r22 = ɵngcc0.ɵɵreference(1); var ctx_r23 = ɵngcc0.ɵɵnextContext(); return ctx_r23.onRowSelectChanged(_r22); })("expandRowChange", function DataTable_tbody_12_Template_tbody_expandRowChange_0_listener() { ɵngcc0.ɵɵrestoreView(_r24); var _r22 = ɵngcc0.ɵɵreference(1); var ctx_r25 = ɵngcc0.ɵɵnextContext(); return ctx_r25.onRowExpandChanged(_r22); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var item_r20 = ctx.$implicit;
    var index_r21 = ctx.index;
    ɵngcc0.ɵɵproperty("item", item_r20)("index", index_r21);
} }
function DataTable_tbody_13_tr_1_td_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "td", 22);
} if (rf & 2) {
    var column_r30 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("hide", !column_r30.visible);
} }
function DataTable_tbody_13_tr_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr");
    ɵngcc0.ɵɵelement(1, "td", 22);
    ɵngcc0.ɵɵelementStart(2, "td", 22);
    ɵngcc0.ɵɵtext(3, "\u00A0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(4, "td", 22);
    ɵngcc0.ɵɵtemplate(5, DataTable_tbody_13_tr_1_td_5_Template, 1, 1, "td", 30);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var index_r28 = ctx.index;
    var ctx_r26 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵclassProp("row-odd", (index_r28 + ctx_r26.items.length) % 2 === 0)("row-even", (index_r28 + ctx_r26.items.length) % 2 === 1);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("hide", !ctx_r26.expandColumnVisible);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("hide", !ctx_r26.indexColumnVisible);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("hide", !ctx_r26.selectColumnVisible);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r26.columns);
} }
function DataTable_tbody_13_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tbody", 28);
    ɵngcc0.ɵɵtemplate(1, DataTable_tbody_13_tr_1_Template, 6, 8, "tr", 29);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    var ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r3.substituteItems);
} }
function DataTable_div_14_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 31);
} }
function DataTable_data_table_pagination_15_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "data-table-pagination");
} }
export var DataTable = (function () {
    function DataTable() {
        this._items = [];
        this.header = true;
        this.pagination = true;
        this.indexColumn = true;
        this.indexColumnHeader = '';
        this.selectColumn = false;
        this.multiSelect = true;
        this.substituteRows = true;
        this.expandableRows = false;
        this.translations = defaultTranslations;
        this.selectOnRowClick = false;
        this.autoReload = true;
        this.showReloading = false;
        this.showDownloadButton = false;
        this._sortAsc = true;
        this._offset = 0;
        this._limit = 10;
        // Reloading:
        this._reloading = false;
        this.reload = new EventEmitter();
        this._displayParams = {}; // params of the last finished reload
        this._scheduledReload = null;
        // Download
        this.download = new EventEmitter();
        // event handlers:
        this.rowClick = new EventEmitter();
        this.rowDoubleClick = new EventEmitter();
        this.headerClick = new EventEmitter();
        this.cellClick = new EventEmitter();
        this.rowExpandChange = new EventEmitter();
        this.selectedRows = [];
        this._selectAllCheckbox = false;
        // column resizing:
        this._resizeInProgress = false;
        this.resizeLimit = 30;
    }
    Object.defineProperty(DataTable.prototype, "items", {
        get: function () {
            return this._items;
        },
        set: function (items) {
            this._items = items;
            this._onReloadFinished();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "sortBy", {
        get: function () {
            return this._sortBy;
        },
        set: function (value) {
            this._sortBy = value;
            this._triggerReload();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "sortAsc", {
        get: function () {
            return this._sortAsc;
        },
        set: function (value) {
            this._sortAsc = value;
            this._triggerReload();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "customSort", {
        get: function () {
            return this._customSort;
        },
        set: function (value) {
            this._customSort = value;
            this._triggerReload();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "offset", {
        get: function () {
            return this._offset;
        },
        set: function (value) {
            this._offset = value;
            this._triggerReload();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "limit", {
        get: function () {
            return this._limit;
        },
        set: function (value) {
            this._limit = value;
            this._triggerReload();
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "page", {
        // calculated property:
        get: function () {
            return Math.floor(this.offset / this.limit) + 1;
        },
        set: function (value) {
            this.offset = (value - 1) * this.limit;
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(DataTable.prototype, "lastPage", {
        get: function () {
            return Math.ceil(this.itemCount / this.limit);
        },
        enumerable: true,
        configurable: true
    });
    // setting multiple observable properties simultaneously
    DataTable.prototype.sort = function (sortBy, asc, customSort) {
        this.sortBy = sortBy;
        this.sortAsc = asc;
        this.customSort = customSort;
    };
    // init
    DataTable.prototype.ngOnInit = function () {
        this._initDefaultValues();
        this._initDefaultClickEvents();
        this._updateDisplayParams();
        if (this.autoReload && this._scheduledReload == null) {
            this.reloadItems();
        }
    };
    DataTable.prototype._initDefaultValues = function () {
        this.indexColumnVisible = this.indexColumn;
        this.selectColumnVisible = this.selectColumn;
        this.expandColumnVisible = this.expandableRows;
    };
    DataTable.prototype._initDefaultClickEvents = function () {
        var _this = this;
        this.headerClick.subscribe(function (tableEvent) { return _this.sortColumn(tableEvent.column); });
        if (this.selectOnRowClick) {
            this.rowClick.subscribe(function (tableEvent) { return tableEvent.row.selected = !tableEvent.row.selected; });
        }
    };
    Object.defineProperty(DataTable.prototype, "reloading", {
        get: function () {
            return this._reloading;
        },
        enumerable: true,
        configurable: true
    });
    DataTable.prototype.reloadItems = function () {
        this._reloading = true;
        this.reload.emit(this._getRemoteParameters());
    };
    DataTable.prototype._onReloadFinished = function () {
        this._updateDisplayParams();
        this._selectAllCheckbox = false;
        this._reloading = false;
    };
    Object.defineProperty(DataTable.prototype, "displayParams", {
        get: function () {
            return this._displayParams;
        },
        enumerable: true,
        configurable: true
    });
    DataTable.prototype._updateDisplayParams = function () {
        this._displayParams = {
            sortBy: this.sortBy,
            customSort: this.customSort,
            sortAsc: this.sortAsc,
            offset: this.offset,
            limit: this.limit
        };
    };
    // for avoiding cascading reloads if multiple params are set at once:
    DataTable.prototype._triggerReload = function () {
        var _this = this;
        if (this._scheduledReload) {
            clearTimeout(this._scheduledReload);
        }
        this._scheduledReload = setTimeout(function () {
            _this.reloadItems();
        });
    };
    DataTable.prototype.downloadItems = function () {
        this.download.emit(this._getRemoteParameters());
    };
    DataTable.prototype.rowClicked = function (row, event) {
        this.rowClick.emit({ row: row, event: event });
    };
    DataTable.prototype.rowDoubleClicked = function (row, event) {
        this.rowDoubleClick.emit({ row: row, event: event });
    };
    DataTable.prototype.headerClicked = function (column, event) {
        if (!this._resizeInProgress) {
            this.headerClick.emit({ column: column, event: event });
        }
        else {
            this._resizeInProgress = false; // this is because I can't prevent click from mousup of the drag end
        }
    };
    DataTable.prototype.cellClicked = function (column, row, event) {
        this.cellClick.emit({ row: row, column: column, event: event });
    };
    // functions:
    DataTable.prototype._getRemoteParameters = function () {
        var params = {};
        if (this.sortBy) {
            params.sortBy = this.sortBy;
            params.customSort = this.customSort;
            params.sortAsc = this.sortAsc;
        }
        if (this.pagination) {
            params.offset = this.offset;
            params.limit = this.limit;
        }
        return params;
    };
    DataTable.prototype.sortColumn = function (column) {
        if (column.sortable) {
            var ascending = this.sortBy === column.property ? !this.sortAsc : true;
            this.sort(column.property, ascending, column.customSort);
        }
    };
    Object.defineProperty(DataTable.prototype, "columnCount", {
        get: function () {
            var count = 0;
            count += this.indexColumnVisible ? 1 : 0;
            count += this.selectColumnVisible ? 1 : 0;
            count += this.expandColumnVisible ? 1 : 0;
            this.columns.toArray().forEach(function (column) {
                count += column.visible ? 1 : 0;
            });
            return count;
        },
        enumerable: true,
        configurable: true
    });
    DataTable.prototype.getRowColor = function (item, index, row) {
        if (this.rowColors !== undefined) {
            return this.rowColors(item, row, index);
        }
    };
    Object.defineProperty(DataTable.prototype, "selectAllCheckbox", {
        get: function () {
            return this._selectAllCheckbox;
        },
        set: function (value) {
            this._selectAllCheckbox = value;
            this._onSelectAllChanged(value);
        },
        enumerable: true,
        configurable: true
    });
    DataTable.prototype._onSelectAllChanged = function (value) {
        this.rows.toArray().forEach(function (row) { return row.selected = value; });
    };
    DataTable.prototype.onRowSelectChanged = function (row) {
        // maintain the selectedRow(s) view
        if (this.multiSelect) {
            var index = this.selectedRows.indexOf(row);
            if (row.selected && index < 0) {
                this.selectedRows.push(row);
            }
            else if (!row.selected && index >= 0) {
                this.selectedRows.splice(index, 1);
            }
        }
        else {
            if (row.selected) {
                this.selectedRow = row;
            }
            else if (this.selectedRow === row) {
                this.selectedRow = row;
            }
        }
        // unselect all other rows:
        if (row.selected && !this.multiSelect) {
            this.rows.toArray().filter(function (row_) { return row_.selected; }).forEach(function (row_) {
                if (row_ !== row) {
                    row_.selected = false;
                }
            });
        }
    };
    DataTable.prototype.onRowExpandChanged = function (row) {
        this.rowExpandChange.emit(row);
    };
    Object.defineProperty(DataTable.prototype, "substituteItems", {
        // other:
        get: function () {
            return Array.from({ length: this.displayParams.limit - this.items.length });
        },
        enumerable: true,
        configurable: true
    });
    DataTable.prototype.resizeColumnStart = function (event, column, columnElement) {
        var _this = this;
        this._resizeInProgress = true;
        drag(event, {
            move: function (moveEvent, dx) {
                if (_this._isResizeInLimit(columnElement, dx)) {
                    column.width = columnElement.offsetWidth + dx;
                }
            },
        });
    };
    DataTable.prototype._isResizeInLimit = function (columnElement, dx) {
        /* This is needed because CSS min-width didn't work on table-layout: fixed.
         Without the limits, resizing can make the next column disappear completely,
         and even increase the table width. The current implementation suffers from the fact,
         that offsetWidth sometimes contains out-of-date values. */
        if ((dx < 0 && (columnElement.offsetWidth + dx) <= this.resizeLimit) || !columnElement.nextElementSibling ||
            (dx >= 0 && (columnElement.nextElementSibling.offsetWidth + dx) <= this.resizeLimit)) {
            return false;
        }
        return true;
    };
    /** @nocollapse */
    DataTable.ctorParameters = function () { return []; };
    DataTable.propDecorators = {
        'items': [{ type: Input },],
        'itemCount': [{ type: Input },],
        'columns': [{ type: ContentChildren, args: [DataTableColumn,] },],
        'rows': [{ type: ViewChildren, args: [DataTableRow,] },],
        'expandTemplate': [{ type: ContentChild, args: ['dataTableExpand',] },],
        'headerTitle': [{ type: Input },],
        'header': [{ type: Input },],
        'pagination': [{ type: Input },],
        'indexColumn': [{ type: Input },],
        'indexColumnHeader': [{ type: Input },],
        'rowColors': [{ type: Input },],
        'rowTooltip': [{ type: Input },],
        'selectColumn': [{ type: Input },],
        'multiSelect': [{ type: Input },],
        'substituteRows': [{ type: Input },],
        'expandableRows': [{ type: Input },],
        'translations': [{ type: Input },],
        'selectOnRowClick': [{ type: Input },],
        'autoReload': [{ type: Input },],
        'showReloading': [{ type: Input },],
        'showDownloadButton': [{ type: Input },],
        'sortBy': [{ type: Input },],
        'sortAsc': [{ type: Input },],
        'customSort': [{ type: Input },],
        'offset': [{ type: Input },],
        'limit': [{ type: Input },],
        'page': [{ type: Input },],
        'reload': [{ type: Output },],
        'download': [{ type: Output },],
        'rowClick': [{ type: Output },],
        'rowDoubleClick': [{ type: Output },],
        'headerClick': [{ type: Output },],
        'cellClick': [{ type: Output },],
        'rowExpandChange': [{ type: Output },],
    };
DataTable.ɵfac = function DataTable_Factory(t) { return new (t || DataTable)(); };
DataTable.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DataTable, selectors: [["data-table"]], contentQueries: function DataTable_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, _c0, true);
        ɵngcc0.ɵɵcontentQuery(dirIndex, DataTableColumn, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.expandTemplate = _t.first);
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.columns = _t);
    } }, viewQuery: function DataTable_Query(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵviewQuery(DataTableRow, true);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.rows = _t);
    } }, inputs: { header: "header", pagination: "pagination", indexColumn: "indexColumn", indexColumnHeader: "indexColumnHeader", selectColumn: "selectColumn", multiSelect: "multiSelect", substituteRows: "substituteRows", expandableRows: "expandableRows", translations: "translations", selectOnRowClick: "selectOnRowClick", autoReload: "autoReload", showReloading: "showReloading", showDownloadButton: "showDownloadButton", items: "items", sortBy: "sortBy", sortAsc: "sortAsc", customSort: "customSort", offset: "offset", limit: "limit", page: "page", itemCount: "itemCount", headerTitle: "headerTitle", rowColors: "rowColors", rowTooltip: "rowTooltip" }, outputs: { reload: "reload", download: "download", rowClick: "rowClick", rowDoubleClick: "rowDoubleClick", headerClick: "headerClick", cellClick: "cellClick", rowExpandChange: "rowExpandChange" }, decls: 16, vars: 12, consts: [[1, "data-table-wrapper"], [4, "ngIf"], [1, "data-table-box"], [1, "table", "table-condensed", "table-bordered", "data-table"], [1, "expand-column-header", 3, "hide"], [1, "index-column-header", 3, "hide"], [3, "textContent"], [1, "select-column-header", 3, "hide"], ["type", "checkbox", 3, "hide", "ngModel", "ngModelChange"], ["class", "column-header", 3, "hide", "sortable", "resizable", "ngClass", "width", "click", 4, "ngFor", "ngForOf"], ["class", "data-table-row-wrapper", "dataTableRow", "", 3, "item", "index", "selectedChange", "expandRowChange", 4, "ngFor", "ngForOf"], ["class", "substitute-rows", 4, "ngIf"], ["class", "loading-cover", 4, "ngIf"], [1, "column-header", 3, "hide", "ngClass", "click"], ["th", ""], [3, "textContent", 4, "ngIf"], [3, "ngTemplateOutlet", "ngOutletContext", 4, "ngIf"], ["class", "column-sort-icon", 4, "ngIf"], ["class", "column-resize-handle", 3, "mousedown", 4, "ngIf"], [3, "ngTemplateOutlet", "ngOutletContext"], [1, "column-sort-icon"], [1, "fa", "fa-sort", "column-sortable-icon", 3, "hide"], [3, "hide"], [1, "fa", "fa-sort-asc", 3, "hide"], [1, "fa", "fa-sort-desc", 3, "hide"], [1, "column-resize-handle", 3, "mousedown"], ["dataTableRow", "", 1, "data-table-row-wrapper", 3, "item", "index", "selectedChange", "expandRowChange"], ["row", ""], [1, "substitute-rows"], [3, "row-odd", "row-even", 4, "ngFor", "ngForOf"], [3, "hide", 4, "ngFor", "ngForOf"], [1, "loading-cover"]], template: function DataTable_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵtemplate(1, DataTable_data_table_header_1_Template, 1, 0, "data-table-header", 1);
        ɵngcc0.ɵɵelementStart(2, "div", 2);
        ɵngcc0.ɵɵelementStart(3, "table", 3);
        ɵngcc0.ɵɵelementStart(4, "thead");
        ɵngcc0.ɵɵelementStart(5, "tr");
        ɵngcc0.ɵɵelement(6, "th", 4);
        ɵngcc0.ɵɵelementStart(7, "th", 5);
        ɵngcc0.ɵɵelement(8, "span", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "th", 7);
        ɵngcc0.ɵɵelementStart(10, "input", 8);
        ɵngcc0.ɵɵlistener("ngModelChange", function DataTable_Template_input_ngModelChange_10_listener($event) { return ctx.selectAllCheckbox = $event; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(11, DataTable_th_11_Template, 7, 14, "th", 9);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(12, DataTable_tbody_12_Template, 2, 2, "tbody", 10);
        ɵngcc0.ɵɵtemplate(13, DataTable_tbody_13_Template, 2, 1, "tbody", 11);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(14, DataTable_div_14_Template, 1, 0, "div", 12);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(15, DataTable_data_table_pagination_15_Template, 1, 0, "data-table-pagination", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.header);
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵproperty("hide", !ctx.expandColumnVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("hide", !ctx.indexColumnVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("textContent", ctx.indexColumnHeader);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("hide", !ctx.selectColumnVisible);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("hide", !ctx.multiSelect)("ngModel", ctx.selectAllCheckbox);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.columns);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.pagination && ctx.substituteRows);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showReloading && ctx.reloading);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.pagination);
    } }, styles: ["[_nghost-%COMP%]     .data-table.table > tbody+tbody {\n    border-top: none;\n}\n[_nghost-%COMP%]     .data-table.table td {\n    vertical-align: middle;\n}\n\n[_nghost-%COMP%]     .data-table > thead > tr > th, [_nghost-%COMP%]     .data-table > tbody > tr > td {\n\toverflow: hidden;\n}\n\n\n[_nghost-%COMP%]     .row-odd {\n    background-color: #F6F6F6;\n}\n[_nghost-%COMP%]     .row-even {\n}\n\n.data-table[_ngcontent-%COMP%]   .substitute-rows[_ngcontent-%COMP%]    > tr[_ngcontent-%COMP%]:hover, [_nghost-%COMP%]     .data-table .data-table-row:hover {\n    background-color: #ECECEC;\n}\n\n\n.data-table[_ngcontent-%COMP%] {\n    box-shadow: 0 0 15px rgb(236, 236, 236);\n    table-layout: fixed;\n}\n\n\n\n.column-header[_ngcontent-%COMP%] {\n    position: relative;\n}\n.expand-column-header[_ngcontent-%COMP%] {\n\twidth: 50px;\n}\n.select-column-header[_ngcontent-%COMP%] {\n\twidth: 50px;\n\ttext-align: center;\n}\n.index-column-header[_ngcontent-%COMP%] {\n\twidth: 40px;\n}\n.column-header.sortable[_ngcontent-%COMP%] {\n    cursor: pointer;\n}\n.column-header[_ngcontent-%COMP%]   .column-sort-icon[_ngcontent-%COMP%] {\n\tfloat: right;\n}\n.column-header.resizable[_ngcontent-%COMP%]   .column-sort-icon[_ngcontent-%COMP%] {\n    margin-right: 8px;\n}\n.column-header[_ngcontent-%COMP%]   .column-sort-icon[_ngcontent-%COMP%]   .column-sortable-icon[_ngcontent-%COMP%] {\n    color: lightgray;\n}\n.column-header[_ngcontent-%COMP%]   .column-resize-handle[_ngcontent-%COMP%] {\n    position: absolute;\n    top: 0;\n    right: 0;\n    margin: 0;\n    padding: 0;\n    width: 8px;\n    height: 100%;\n    cursor: col-resize;\n}\n\n\n\n.data-table-box[_ngcontent-%COMP%] {\n    position: relative;\n}\n\n.loading-cover[_ngcontent-%COMP%] {\n   position: absolute;\n   width: 100%;\n   height: 100%;\n   background-color: rgba(255, 255, 255, 0.3);\n   top: 0;\n}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DataTable, [{
        type: Component,
        args: [{
                selector: 'data-table',
                template: TABLE_TEMPLATE,
                styles: [TABLE_STYLE]
            }]
    }], function () { return []; }, { header: [{
            type: Input
        }], pagination: [{
            type: Input
        }], indexColumn: [{
            type: Input
        }], indexColumnHeader: [{
            type: Input
        }], selectColumn: [{
            type: Input
        }], multiSelect: [{
            type: Input
        }], substituteRows: [{
            type: Input
        }], expandableRows: [{
            type: Input
        }], translations: [{
            type: Input
        }], selectOnRowClick: [{
            type: Input
        }], autoReload: [{
            type: Input
        }], showReloading: [{
            type: Input
        }], showDownloadButton: [{
            type: Input
        }], reload: [{
            type: Output
        }], download: [{
            type: Output
        }], rowClick: [{
            type: Output
        }], rowDoubleClick: [{
            type: Output
        }], headerClick: [{
            type: Output
        }], cellClick: [{
            type: Output
        }], rowExpandChange: [{
            type: Output
        }], items: [{
            type: Input
        }], sortBy: [{
            type: Input
        }], sortAsc: [{
            type: Input
        }], customSort: [{
            type: Input
        }], offset: [{
            type: Input
        }], limit: [{
            type: Input
        }], page: [{
            type: Input
        }], itemCount: [{
            type: Input
        }], columns: [{
            type: ContentChildren,
            args: [DataTableColumn]
        }], rows: [{
            type: ViewChildren,
            args: [DataTableRow]
        }], expandTemplate: [{
            type: ContentChild,
            args: ['dataTableExpand']
        }], headerTitle: [{
            type: Input
        }], rowColors: [{
            type: Input
        }], rowTooltip: [{
            type: Input
        }] }); })();
    return DataTable;
}());
